controllers/users.js

```diff
@@ -1096,6 +1103,26 @@ const updateUsernames = async (req, res) => {
   }
 };
 
+const updateProfile = async (req, res) => {
+  try {
+    const { id: currentUserId, roles = {} } = req.userData;
+    const isSelf = req.params.userId === currentUserId;
+    const isSuperUser = roles[ROLES.SUPERUSER];
+    const profile = req.query.profile === "true";
+
+    if (isSelf && profile && req.query.dev === "true") {
+      return await updateSelf(req, res);
+    } else if (isSuperUser) {
+      return await updateUser(req, res);
+    }
+
+    return res.boom.unauthorized("You are not authorized for this action.");
```

Comment:
If isSelf and profile are true and dev is false which means it is a bad request,  then why are we sending 401 response?

Reply:
done thanx!